keys/src/cashaddress.rs

```diff
@@ -0,0 +1,490 @@
+use std::str::FromStr;
+use std::fmt;
+
+const DEFAULT_PREFIX: NetworkPrefix = NetworkPrefix::BitcoinCash;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum AddressType {
+    /// Pay to PubKey Hash
+	/// https://bitcoin.org/en/glossary/p2pkh-address
+    P2PKH,
+    /// Pay to Script Hash
+    /// https://bitcoin.org/en/glossary/p2sh-address
+    P2SH,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum NetworkPrefix {
+    BitcoinCash,
+    BchTest,
+    BchReg,
+}
+
+impl fmt::Display for NetworkPrefix {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let as_str = match self {
+            NetworkPrefix::BitcoinCash => "bitcoincash",
+            NetworkPrefix::BchTest => "bchtest",
+            NetworkPrefix::BchReg => "bchreg",
+        };
+        write!(f, "{}", as_str)
+    }
+}
+
+impl FromStr for NetworkPrefix {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let s = s.to_lowercase();
+        let prefix = match s.as_str() {
+            "bitcoincash" => NetworkPrefix::BitcoinCash,
+            "bchtest" => NetworkPrefix::BchTest,
+            "bchreg" => NetworkPrefix::BchReg,
+            _ => return Err("Unexpected network prefix".into()),
+        };
+        Ok(prefix)
+    }
+}
+
+impl From<&str> for NetworkPrefix {
+    fn from(s: &str) -> Self {
+        s.parse().unwrap()
```

Comment:
Is it safe to simply unwrap here? I think unwrap may be fine for conversion from `&'static str`, but from arbitrary str we can get panic easily if we get unexpected input.

Reply:
What's better way? Move the impl From<&str> to tests module? Or remove it at all?